src/vcpkg/archives.cpp

```diff
@@ -127,38 +127,77 @@ namespace
     }
 #endif // ^^^ _WIN32
 
-    void extract_archive_to_empty(const Filesystem& fs,
-                                  const ToolCache& tools,
-                                  MessageSink& status_sink,
-                                  const Path& archive,
-                                  const Path& to_path)
+}
+
+namespace vcpkg
+{
+    static ExtractionType guess_extraction_type(const Path& archive)
+
```

Comment:
I don't know if it would be meaningful to add test cases for this function since it essentially just calls `Path::extension()`. Additionally, this isn't new. I just sucked out the logic that was previously in `extract_archive` into its own function. The logic is exactly the same. It just returns an enum. I can add the test cases if you still think it's necessary. :)